binlog_streamer.go

```diff
@@ -76,22 +77,31 @@ func (s *BinlogStreamer) createBinlogSyncer() error {
 }
 
 func (s *BinlogStreamer) ConnectBinlogStreamerToMysql() error {
-	err := s.createBinlogSyncer()
+	s.ensureLogger()
+
+	currentPosition, err := ShowMasterStatusBinlogPosition(s.DB)
 	if err != nil {
+		s.logger.WithError(err).Error("failed to read current binlog position")
 		return err
 	}
 
-	s.logger.Info("reading current binlog position")
-	s.lastStreamedBinlogPosition, err = ShowMasterStatusBinlogPosition(s.Db)
+	return s.ConnectBinlogStreamerToMysqlFrom(currentPosition)
+}
+
+func (s *BinlogStreamer) ConnectBinlogStreamerToMysqlFrom(startFromBinlogPosition mysql.Position) error {
+	s.ensureLogger()
+
+	err := s.createBinlogSyncer()
 	if err != nil {
-		s.logger.WithError(err).Error("failed to read current binlog position")
 		return err
 	}
 
+	s.lastStreamedBinlogPosition = startFromBinlogPosition
```

Comment:
Having it set there would be kind of redundant as this is an internal tracking variable for the main BinlogStreamer loop. Basically, it would have to look like:

```
ConnectBinlogStreamerToMySqlFrom():
  s.startFromBinlogPosition = startFromBinlogPosition

Run():
  s.lastStreamedBinlogPosition = s.startFromBinlogPosition
  for ... s.lastStreamedBinlogPosition.Compare(s.targetBinlogPosition) < 0:
     ....
```

Without setting lastStreamedBinlogPosition before we start that loop, the conditional would be more complex.

Note that this variable has nothing to do with the StateTracker binlog position, which is set in BinlogWriter and once in Ferry during initial connection.